File: app/src/main/java/com/alorma/contactnotes/data/contacts/store/ContactDao.java
package com.alorma.contactnotes.data.contacts.store;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface ContactDao {
    @Query("SELECT * FROM contacts")
    Flowable<List<ContactEntity>> getAll();

    @Query("SELECT * FROM contacts WHERE id LIKE :id LIMIT 1")
    ContactEntity findById(String id);

    @Insert
    void insertAll(ContactEntity... users);

    @Delete
    void delete(ContactEntity user);
}

File: app/src/main/java/com/alorma/contactnotes/data/contacts/store/ContactDao.kt
package com.alorma.contactnotes.data.contacts.store

import android.arch.persistence.room.*

@Dao
interface ContactDao {
    @get:Query("SELECT * FROM contacts")
    val all: List<ContactEntity>

    @Query("SELECT * FROM contacts WHERE id LIKE :id LIMIT 1")
    fun findById(id: String): ContactEntity

    @Query("SELECT * FROM contacts WHERE androidId LIKE :id LIMIT 1")
    fun findByAndroidId(id: String): ContactEntity

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(vararg contactEntity: ContactEntity)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun update(vararg contactEntity: ContactEntity)

    @Delete
    fun delete(user: ContactEntity)
}